components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -6,16 +6,17 @@ package mozilla.components.support.utils
 
 import android.net.Uri
 import android.text.TextUtils
+import java.util.regex.Pattern
 
 object URLStringUtils {
     fun isURLLike(string: String, safe: Boolean = false) =
-        if (safe) {
-            string.matches(WebURLFinder.fuzzyUrlRegex)
-        } else {
-            string.matches(WebURLFinder.fuzzyUrlNonWebRegex)
-        }
+            if (safe) {
+                string.matches(WebURLFinder.fuzzyUrlRegex)
+            } else {
+                string.matches(WebURLFinder.fuzzyUrlNonWebRegex)
+            }
 
-    fun isSearchTerm(string: String) = !isURLLike(string, false)
+    fun isSearchTerm(string: String) = !isURLLenient.matcher(string).matches()
```

Comment:
@hawkinsw Thanks for taking this on! 

I am curious what the plan is in Fenix. Is it to switch these `isUrl` calls to `!isSearchTerm` (since that's the faster method now but currently has no caller): https://github.com/mozilla-mobile/fenix/search?utf8=✓&q=isUrl()

We then should also update the `ToolbarInteractor` in A-C to get the benefits: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarInteractor.kt#L29

We could also have a `String.isUrlStrict` and `String.isUrl` and only use the strict method where needed.

Reply:
> @hawkinsw Thanks for taking this on!
> 
> I am curious what the plan is in Fenix. Is it to switch these `isUrl` calls to `!isSearchTerm` (since that's the faster method now but currently has no caller)? [https://github.com/mozilla-mobile/fenix/search?utf8=✓&q=isUrl()](https://github.com/mozilla-mobile/fenix/search?utf8=%E2%9C%93&q=isUrl())

That is exactly the path that I thought we would take. However, see below ...
> 
> We then should also update the `ToolbarInteractor` in A-C to get the benefits: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarInteractor.kt#L29
> 

Thanks for pointing this out. I will push an updated PR to address this! Thanks again!
> We could also have a `String.isUrlStrict` and `String.isUrl` and only use the strict method where needed.

I actually like this better. I am sending up an update to this PR that addresses this and the previous point. Thanks for the review. I will rerequest a review as soon as it is up!